### PR TITLE
Ft blog #139551349

### PR DESCRIPTION
**Pull request by Gathondu**
#### What does this PR do?
Adds a blog functionality to hc

### CODE REVIEW

File: hc/api/models.py [hunk 1]
```diff
@@ -267,3 +267,18 @@ class Meta:
...
+
+
+class Blog(models.Model):
+
+    class Meta:
+        get_latest_by = "created"
+
+    name = models.CharField(max_length=100, default="Blog Title")
+    tags = models.CharField(max_length=500, blank=True)
+    user = models.ForeignKey(User, blank=True, null=True)
```
Comment (osule) [inline hc/api/models.py]: A blog post should always have an author. We can rename this field to `author`.

Comment (Gathondu) [reply hc/api/models.py]: i've refractored the model file

Comment (Gathondu) [reply hc/api/models.py]: i've refractored that model class

Comment (Gathondu) [reply hc/api/models.py]: i hadn't pushed the changes

File: hc/api/models.py [hunk 2]
```diff
@@ -267,3 +267,18 @@ class Meta:
...
+    created = models.DateTimeField(auto_now_add=True)
+    content = models.CharField(max_length=2000, default="no content")
+
+    def tags_list(self):
+        return [t.strip() for t in self.tags.split(" ") if t.strip()]
```
Comment (osule) [inline hc/api/models.py]: Is there a reason for using `if t.strip()`

File: hc/front/urls.py [hunk 3]
```diff
@@ -35,7 +40,9 @@
     url(r'^checks/([\w-]+)/', include(check_urls)),
     url(r'^integrations/', include(channel_urls)),
     url(r'^failed/$', views.failed_checks, name="hc-failed-checks"),
-
+    url(r'^blog/$', views.blog, name="hc-blogs"),
```
Comment (osule) [inline hc/front/urls.py]: Can we move these 2 URLs to the `blog_urls` list?

File: hc/front/views.py [hunk 4]
```diff
@@ -575,3 +575,55 @@ def failed_checks(request):
     }
 
     return render(request, "front/failed_checks.html", ctx)
+
+
+@login_required
+def blog(request):
+    b = Blog.objects.filter(user=request.team.user).order_by("created")
+    blogs = list(b)
```
Comment (osule) [inline hc/front/views.py]: `querysets` are iterable. We should be able to iterate over them without explicitly converting to a list

File: hc/front/views.py [hunk 5]
```diff
@@ -575,3 +575,55 @@ def failed_checks(request):
...
+    tags = [blog.tags_list() for blog in blogs]
+
+    ctx = {
```
Comment (osule) [inline hc/front/views.py]: Great idea. Can we spell this out as context?

File: hc/front/views.py [hunk 6]
```diff
@@ -575,3 +575,55 @@ def failed_checks(request):
...
+
+@login_required
+def new_blog(request):
+    return render(request, "front/new_blog.html")
+
+
+@login_required
+@uuid_or_400
+def update_blog(request, name):
+    assert request.method == "POST"
```
Comment (osule) [inline hc/front/views.py]: Can we use an if here?

File: hc/front/views.py [hunk 7]
```diff
@@ -575,3 +575,55 @@ def failed_checks(request):
...
+        blog.tags = form.cleaned_data["tags"]
+        blog.save()
+
+    return redirect("hc-blogs")
+
+
+@login_required
+@uuid_or_400
+def remove_blog(request, name):
+    assert request.method == "POST"
```
Comment (osule) [inline hc/front/views.py]: Same with here. Lets use an `if`

File: hc/api/models.py (hunk 2 above)
Comment (Gathondu) [reply hc/api/models.py]: incase the user had input alot of spaces

File: hc/front/views.py (hunk 4 above)
Comment (Gathondu) [reply hc/front/views.py]: ok

File: hc/api/models.py [hunk 8]
```diff
@@ -267,3 +267,18 @@ class Meta:
     channel = models.ForeignKey(Channel)
     created = models.DateTimeField(auto_now_add=True)
     error = models.CharField(max_length=200, blank=True)
+
+
+class Blog(models.Model):
```
Comment (osule) [inline hc/api/models.py]: We should rename this model to BlogPost. This is what it essentially is.

File: hc/front/views.py (hunk 5 above)
Comment (Gathondu) [reply hc/front/views.py]: ok

File: hc/api/models.py [hunk 9]
```diff
@@ -267,3 +267,47 @@ class Meta:
     channel = models.ForeignKey(Channel)
     created = models.DateTimeField(auto_now_add=True)
     error = models.CharField(max_length=200, blank=True)
+
+
+class PublishedManager(models.Manager):
```
Comment (osule) [inline hc/api/models.py:272]: Noice!

Comment (osule) [reply hc/api/models.py:272]: Manage all the things!

File: hc/api/models.py [hunk 10]
```diff
@@ -267,3 +267,47 @@ class Meta:
...
+    updated = models.DateTimeField(auto_now=True)
+    status = models.CharField(max_length=10, choices=STATUS_CHOICES,
+                              default='draft')
+    tags = models.CharField(max_length=500, blank=True)
+
+    objects = models.Manager()
+    published = PublishedManager()
+
+    def tags_list(self):
+        return [t.strip() for t in self.tags.split(" ") if t.strip()]
```
Comment (osule) [inline hc/api/models.py:302]: See my comment about `if t.strip()`

File: hc/front/forms.py [hunk 11]
```diff
@@ -41,3 +43,46 @@ class AddWebhookForm(forms.Form):
...
+
+        for part in self.cleaned_data["tags"].split(" "):
+            part = part.strip()
+            if part != "":
+                l.append(part)
+
+        return " ".join(l)
+
+
+# class AddPostForm(forms.Form):
```
Comment (osule) [inline hc/front/forms.py:65]: We'll want to clean this up.

**Review by osule — CHANGES_REQUESTED** [review]
Please add tests

File: hc/api/models.py (hunk 10 above)
Comment (osule) [reply hc/api/models.py:302]: Scratch this

**Pull request closed without merging**